test/FunctionalTests/Microsoft.OData.Core.Tests/ODataNotificationStreamTests.cs

```diff
@@ -32,7 +32,7 @@ public void NotificationStreamDisposeShouldInvokeStreamDisposed(bool synchronous
             // We care about the notification stream being disposed
             // We don't care about the stream passed to the notification stream
             using (var notificationStream = new ODataNotificationStream(
-                new MemoryStream(),
+                this.stream,

```

Comment:
Why this change necessary? Is `this.stream` used in the test? Do you make a check to verify that it was not disposed?

Reply:
Please observe that I'm verifying that `StreamDisposed` or `StreamDiposedAsync` are invoked by reading the content of the stream they wrote to when they were invoked. In the previous case, the two methods were writing to a stream entirely different to the one passed to `ODataNotificationStream` constructor. Since `ODataNotifictionStream` is not disposing the stream after the change, the same stream being passed to the constructor is the same one that `StreamDisposed`/`StreamDiposedAsync` are writing to, with the assurance that it'll still be available for the  the helper `ReadStreamContents`/`ReadStreamContentsAsync` methods to read from. That is why I noted in the PR description that the change I made to the tests is the confirmation needed to verify that the stream is not disposed